.github/workflows/network-operator-docker-and-helm-ci.yaml

```diff
@@ -40,29 +45,34 @@ jobs:
   helm-package-publish:
     needs: docker-build-push
     runs-on: ubuntu-latest
+    env:
+      NGC_REPO: nvstaging/mellanox/network-operator
     steps:
-    - uses: actions/checkout@v4
-      with:
-        fetch-tags: true
-    - name: Determine versions
+    - if: github.ref_name == env.STABLE_BRANCH || github.ref_type == 'tag'
+      uses: actions/checkout@v4
+    - if: github.ref_name == env.STABLE_BRANCH
+      name: Determine versions (when git branch)
       run: |
-        git_tag=$(git describe --tags --exact-match 2>/dev/null | grep -E '${{ env.VALID_TAG_REGEX}}' || true)  # git tag, if it exists
-        git_sha=$(git rev-parse --short HEAD)  # short git commit hash
-        chart_current_version=$(yq '.version' deployment/network-operator/Chart.yaml)
-
-        if [ -n "$git_tag" ]; then
-          app_version=$git_tag
-          chart_version=${git_tag:1}  # without the 'v' prefix
-        elif [ $(git branch --show-current) == "master" ]; then
-          app_version=$git_sha
-          chart_version=$chart_current_version-$git_sha
-        fi
-
+        app_version=$(git rev-parse --short HEAD)  # short git commit hash
+        current_chart_version=$(yq '.version' deployment/network-operator/Chart.yaml)
+        echo APP_VERSION=""$app_version""                    >> $GITHUB_ENV
+        echo VERSION=""$current_chart_version-$app_version"" >> $GITHUB_ENV
+    - if: github.ref_type == 'tag'
+      name: Determine versions (when git tag)
+      run: |
+        git_tag=${{ github.ref_name }}
+        app_version=$git_tag
+        chart_version=${git_tag:1}  # without the 'v' prefix
```

Comment:
(Can do after the merge, just so we don't forget) Need to make a test release to check that all versions in the helm chart / image versions are correct